### DISCUSSION TITLE
Can i use other tree algorithms for online anomaly detection?

### DISCUSSION THREAD

**Discussion by ercangunbilek** [online-ml/river#993] [Q&A] [answered]
Such as Hoeffding Tree , extremly Fast desicion tree ?

**Comment by MaxHalford** [comment, marked as answer]
Hey. The algorithms you mentioned are supervised, so the short answer is no. But there is the `HalfSpaceTrees` anomaly detector which works reasonably well. Have you tried it?

The long answer is that you can use any supervised algorithm to do anomaly detection. The trick is to use any of the features in the dataset as the target. If you're lucky the predicted output will be correlated with what an anomaly detection model would have given you. But here be dragons, and I would only venture here if regular anomaly detection methods haven't worked.

**Reply by ercangunbilek** [reply]
Actually I want to develop a new method on HST. I'm thinking how can I improve. I'm looking for a different way, like using different trees in HST algorithm.What can i do ? Can you suggest  something about anomaly detection on streaming data ? Maybe, I can contribute river library.

**Reply by MaxHalford** [reply]
If I understand correctly, you would like to improve HST? If so, then be my guest :). I suggest checking out [this](https://www.notion.so/maxhalford/RS-Forest-for-anomaly-detection-d92bbad9085f44669d9116225786efb3) card from our roadmap.
